### ISSUE TITLE
Remove all subscribers when doing rmpsp

### ISSUE THREAD

**Issue by strong84** [uniqush/uniqush-push#245]
When remove a psp(rmpsp), should it also remove all the subscribers belongs to this psp?

In my uniqush push server, when I remove a psp, all the subscribers are remained.

**Comment by TysonAndre** [comment]
It's currently not designed to remove all of the subscribers belonging to the psp. This feature would be nice to have (e.g. add `?purgesubscribers=1`)

- Making it opt-in would make sense, to avoid accidentally deleting subscribers when a user is trying to change an existing PSP.
- Currently, uniqush-push does not require SCAN or KEYS for core functionality (except for listing all subscribers with wildcard operators). Some uniqush-push installations may use a proxy that forwards commands to multiple different redis instances, and SCAN/KEYS wouldn't be available for those proxies.
- This will have common functionality with https://github.com/uniqush/uniqush-push/issues/191 - both require iterating over delivery points and associating them with subscriptions
- The rate of deletion should be throttled to avoid impacting regular push notifications and other applications using the same redis instance. There should be a way to view progress and to cancel an in-progress task to remove all subscribers.

**Comment by strong84** [comment]
Cool!
BTW, I notice the api document is not up to date on http://uniqush.org/documentation/usage.html
new features are buried in github release notes. this could lose track. and difficult for new user like me.

**Comment by strong84** [comment]
> BTW, I notice the api document is not up to date on http://uniqush.org/documentation/usage.html
> new features are buried in github release notes. this could lose track. and difficult for new user like me.

Just discovered [NEWS.md](https://github.com/uniqush/uniqush-push/blob/master/NEWS.md) in the repo. all the changes are listed there.

**Comment by mishan** [comment]
No, and that's intentional: `/rmpsp` removes the provider and leaves the subscriptions, so that a provider can be re-added (or replaced with `replace=true`, new in 2.8.0) without every device re-subscribing. 2.8.0 also fixed a bug where a *read* after `/rmpsp` deleted subscriptions, which may have been what you were seeing the opposite of. Removing all of a subscriber's devices in one call is tracked in #216.
